soroban-sdk-macros/src/lib.rs

```diff
@@ -95,6 +97,45 @@ pub fn contractimpl(_metadata: TokenStream, input: TokenStream) -> TokenStream {
     }
 }
 
+#[derive(Debug, FromMeta)]
+struct MetadataArgs {
+    key: String,
+    val: String,
+}
+
+#[proc_macro]
+pub fn metadata(metadata: TokenStream) -> TokenStream {
+    let args = parse_macro_input!(metadata as AttributeArgs);
+    let args = match MetadataArgs::from_list(&args) {
+        Ok(v) => v,
+        Err(e) => return e.write_errors().into(),
+    };
+
+    let gen = {
+        //TODO: is key necessary?
```

Comment:
What's your thinking around making it optional / not used?

Reply:
Depends on what we do to handle the collisions mentioned below. If we only allow one metadata variable, then the key isn't necessary.

Reply:
The variable name won't end up in the custom section, only the contents. And the contents of all the key-value pairs will  be concatenated together in potentially random order. So I think we need the key still.

Reply:
Well the section name itself (`contractmetav0 `) will "collide" if the macro is used more than once right?

Reply:
I was thinking about appending the key to `contractmetav0` as well.

Reply:
Oh wow I thought I tested this but looks like I didn't regenerate the wasm file. Looks like you're right.